### DISCUSSION TITLE
issue: Conditional Controller field value may not reflect actual value of form

### DISCUSSION THREAD

**Discussion by Fyers** [react-hook-form/react-hook-form#10225] [Q&A]
### Version Number

7.43.8

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-get-started-forked-ypy5sw?file=/src/index.js

### Steps to reproduce

1. Change value of input field. Only then the controller field value is shown correctly
2. Or: Switch lines 17 and 18 (This will result in different values to be shown inside the controller)

### Expected behaviour

The Controller should always have the correct value. Seems like a race condition since switching line 17 and 18 will change the behaviour.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
I am not quite sure about your usage, but your controller is missing `defaultValues` at `useForm`.

**Comment by Fyers** [comment]
> I am not quite sure about your usage, but your controller is missing `defaultValues` at `useForm`.

In my usecase there are no default values that I can use for `useForm` until after some time the fetch was completed and the form will be reset with the default values. If i render everything after the fetch was completed it works but the form in my case should be visible with a loading indicator.

**Comment by bluebill1049** [comment]
then you should keep inline defaultValue at least, we do not support both for them are missing.
